File: modules/angular2/src/core/forms/directives/control_container.ts
import {Form} from './form_interface';
import {AbstractControlDirective} from './abstract_control_directive';

/**
 * A directive that contains multiple {@link NgControl}s.
 *
 * Only used by the forms module.
 */
export class ControlContainer extends AbstractControlDirective {
  name: string;

  /**
   * Get the form to which this container belongs.
   */
  get formDirective(): Form { return null; }

  /**
   * Get the path to this container.
   */
  get path(): string[] { return null; }
}

File: lib/src/common/forms/directives/control_container.dart
library angular2.src.common.forms.directives.control_container;

import "form_interface.dart" show Form;
import "abstract_control_directive.dart" show AbstractControlDirective;

/**
 * A directive that contains multiple [NgControl]s.
 *
 * Only used by the forms module.
 */
class ControlContainer extends AbstractControlDirective {
  String name;
  /**
   * Get the form to which this container belongs.
   */
  Form get formDirective {
    return null;
  }

  /**
   * Get the path to this container.
   */
  List<String> get path {
    return null;
  }
}
